src/EditorFeatures/Core/ReferenceHighlighting/ReferenceHighlightingViewTaggerProvider.cs

```diff
@@ -163,7 +163,8 @@ protected override Task ProduceTagsAsync(TaggerContext<NavigableHighlightTag> co
             var document = documentHighlights.Document;
 
             var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
-            var textSnapshot = text.FindCorrespondingEditorTextSnapshot();
+
+            var textSnapshot = context.SpansToTag.FirstOrDefault(s => s.Document == document).SnapshotSpan.Snapshot;
             if (textSnapshot == null)
```

Comment:
what happened here?

Reply:
So for normal open documents, while the file is open the document text is directly linked to a text snapshot. In the case of generated document it's a bit murkier -- they're _not_ connected to a snapshot if we're just using text that was generated in the background. If the background-computed text matches the open text, I was reusing the existing snapshot rather than forking it with identical text. So in the snapshot you get FindCorrespondingEditorTextSnapshot may not work. I _could_ fork it anyways but such a fork would throw away the red nodes for tree parsed for the source generated file. Rather than using that helper, we're already given a list of documents and corresponding snapshots, so we can just use that directly.